numpy/polynomial/polynomial.py

```diff
@@ -1336,6 +1337,89 @@ def polyfit(x, y, deg, rcond=None, full=False, w=None):
     return pu._fit(polyvander, x, y, deg, rcond, full, w)
 
 
+def polyfit2d(x, y, z, deg=1, rcond=None, full=False, w=None, max_degree=None, scale=True):
+    """A simple 2D polynomial fit to data x, y, z
+    The polynomial can be evaluated with numpy.polynomial.polynomial.polyval2d
+
+    Parameters
+    ----------
+    x : array_like
+        x coordinates
+    y : array_like
+        y coordinates
+    z : array_like
+        data values
+    deg : {int, 2-tuple}, optional
+        degree of the polynomial fit in x and y direction, by default 1
+    rcond : float, optional
+        Relative condition number of the fit.  Singular values smaller
+        than `rcond`, relative to the largest singular value, will be
+        ignored.  The default value is ``len(x)*eps``, where `eps` is the
+        relative precision of the platform's float type, about 2e-16 in
+        most cases.
+    full : bool, optional
+        Switch determining the nature of the return value.  When ``False``
+        (the default) just the coefficients are returned; when ``True``,
+        diagnostic information from the singular value decomposition (used
+        to solve the fit's matrix equation) is also returned.
+    w : array_like, shape (`M`,), optional
+        Weights. If not None, the contribution of each point
+        ``(x[i],y[i])`` to the fit is weighted by `w[i]`. Ideally the
+        weights are chosen so that the errors of the products ``w[i]*y[i]``
+        all have the same variance. The default value is None.
+    max_degree : int, optional
+        If given the maximum combined degree of the coefficients is limited
+        to this value, i.e. all terms with `n` + `m` > max_degree are set to 0.
+        The default is None.
+    scale : bool, optional
```

Comment:
Scaling is even more important for the other polynomial basis whose weights have compact support, which is why I'd like to see a class for 2-d objects at some point so that the extra information, also including the selection of coefficients, could be carried along. I don't think it belongs in the base fitting function, however. It is also possible to scale the 2d domain to `[-1, 1]` in each direction and get good results in most cases, images and such.